src/components/pages/bc-send-payment.ts

```diff
@@ -39,6 +40,38 @@ export class SendPayment extends withTwind()(BitcoinConnectElement) {
   })
   invoice?: string;
 
+  @state()
+  _fiatValue?: string;
+
+  private async fetchFiatValue() {
+    const invoice = this._invoice || this.invoice;
+    if (!invoice) {
+      return;
+    }
+
+    try {
+      const decodedInvoice = new Invoice({ pr: invoice });
+      const fiatValue = await fiat.getFiatValue({
+        satoshi: decodedInvoice.satoshi,
+        currency: 'usd',
+      });
+
+      this._fiatValue = fiatValue.toLocaleString(undefined, {
+        style: 'currency',
+        currency: 'USD',
+      });
+    } catch (error) {
+      console.error('Error fetching fiat value:', error);
+      this._fiatValue = 'Error fetching fiat value';
+    }
+  }
+
+  override connectedCallback() {
+    super.connectedCallback();
+    this.fetchFiatValue();
```

Comment:
I don't think we can fetch the fiat value here as it won't update if the invoice changes